### PR TITLE
hybrid-array: add slice_as_chunks(_mut) methods, minor refactoring

### PR DESCRIPTION
**Pull request by newpavlov**
As discussed [here](https://github.com/RustCrypto/traits/pull/1358#discussion_r1373973004) these functions will allow us to remove a bit of `unsafe` from downstream crates and it will be useful for users who would like to work with blocks of data.

### CODE REVIEW

File: hybrid-array/src/lib.rs [hunk 1]
```diff
@@ -13,7 +13,6 @@
     clippy::cast_sign_loss,
     clippy::checked_conversions,
     clippy::implicit_saturating_sub,
-    clippy::arithmetic_side_effects,
```
Comment (tarcieri) [inline hybrid-array/src/lib.rs:16]: I'd prefer to keep this on at a crate-level. Perhaps you can disable it at the individual function level or operation level with a `SAFETY`-style comment about why overflow/underflow won't occur?

**Comment by tarcieri** [comment]
#951 seems somewhat related, though not something I'm suggesting we necessarily tackle in this PR

**Review by tarcieri — APPROVED** [review]

**Comment by newpavlov** [comment]
>#951 seems somewhat related, though not something I'm suggesting we necessarily tackle in this PR

Yeah, we can do it in a separate PR. I wonder if it can be handled by `From`/`Into` impls.

**Comment by tarcieri** [comment]
@newpavlov that's the sort of line of thinking I was curious about, i.e. how to compose various types of (reference) conversions

**Pull request merged**